### ISSUE TITLE
M5/mal tests tst1012, tst1014, tst601 hang with assertions enabled since recent (Fri, Jan 25 2013) checkins

### ISSUE THREAD

**Issue by monetdb-team** [MonetDB/MonetDB#3226]
Date: 2013-01-28 08:28:46 +0100
From: @drstmane
To:   MonetDB5 devs &lt;<bugs-monetdb5>&gt;
Version: -- development

Last updated: 2013-01-30 08:34:45 +0100

## Comment 18437
Date: 2013-01-28 08:28:46 +0100
From: @drstmane

Only when assertions are enabled, tests
 monetdb5/mal/Tests/tst1012.mal
 monetdb5/mal/Tests/tst1014.mal
 monetdb5/mal/Tests/tst601.mal
 appear to hang since recent (Friday January 25 2013) checkins; cf.,
 
 http://monetdb.cwi.nl/testweb/web/testgrid.php?serial=46771:f024941411ed&order=platform,arch,compiler
 
 http://monetdb.cwi.nl/testweb/web/showtestoutput.php?serial=46771:f024941411ed&target=GNU-Fedora-x86_64-propcheck-assert&module=monetdb5&test=monetdb5%2Fmal%2Ftst1012&which=err
 
 http://monetdb.cwi.nl/testweb/web/testchange.php?test=46771:f024941411ed/GNU-Fedora-x86_64-propcheck-assert/monetdb5/mTests/monetdb5%2Fmal%2Ftst1012
 
 http://monetdb.cwi.nl/testweb/web/commits.php?serial=46753:295a2a015d53

## Comment 18439
Date: 2013-01-29 10:06:31 +0100
From: @sjoerdmullender

Changeset [1c20086be294](https://dev.monetdb.org/hg/MonetDB?cmd=changeset;node=1c20086be294)  made by Sjoerd Mullender <sjoerd@acm.org> in the MonetDB repo, refers to this bug.
 
 For complete details, see [http//devmonetdborg/hg/MonetDB?cmd=changeset;node=1c20086be294](https://dev.monetdb.org/hg/MonetDB?cmd=changeset;node=http//devmonetdborg/hg/MonetDB?cmd=changeset;node=1c20086be294) 
 
 Changeset description:
 
 	After a lock has been destroyed, it may not be used anymore.
 	In other words, the lock was destroyed prematurely.
 	This fixed bug #3226 once the fix is propagated to the default branch.

## Comment 18451
Date: 2013-01-30 08:34:45 +0100
From: @drstmane

Nightly testing confirms fix. Closing.
